cmd/kubectl-gadget/collector.go

```diff
@@ -0,0 +1,389 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/k8sutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+var collectorCmd = &cobra.Command{
+	Use:   "process-collector",
+	Short: "Collect processes",
+	Run:   collectorCmdRun("process-collector"),
+}
+
+var (
+	collectorParamLabel         string
+	collectorParamNode          string
+	collectorParamPodname       string
+	collectorParamContainername string
+	collectorParamAllNamespaces bool
+	collectorParamJsonOutput    bool
+	collectorParamThreads       bool
+)
+
+const (
+	GADGET_OPERATION = "gadget.kinvolk.io/operation"
+)
+
+func init() {
+	commands := []*cobra.Command{
+		collectorCmd,
+	}
+
+	// Add flags for all collector gadgets
+	for _, command := range commands {
+		rootCmd.AddCommand(command)
+		command.PersistentFlags().StringVarP(
+			&collectorParamLabel,
+			"selector",
+			"l",
+			"",
+			fmt.Sprintf("Labels selector to filter on. Only '=' is supported (e.g. key1=value1,key2=value2)."),
+		)
+
+		command.PersistentFlags().StringVar(
+			&collectorParamNode,
+			"node",
+			"",
+			fmt.Sprintf("Show only events from pods running in that node"),
+		)
+
+		command.PersistentFlags().StringVarP(
+			&collectorParamPodname,
+			"podname",
+			"p",
+			"",
+			fmt.Sprintf("Show only events from pods with that name"),
+		)
+
+		command.PersistentFlags().StringVarP(
+			&collectorParamContainername,
+			"containername",
+			"c",
+			"",
+			fmt.Sprintf("Show only events from containers with that name"),
```

Comment:
```suggestion
			fmt.Sprintf("Show only processes from containers with that name"),
```

Same in other places?